web/src/components/HomePage/Home.jsx

```diff
@@ -1,24 +1,74 @@
 "use client";
-import PlantCard from "../PlantCard/PlantCard";
-import { useEffect, useState } from "react";
+import { useState, useEffect, useMemo } from "react";
 import styles from "./page.module.css";
+import SearchFilterSort from "@/components/SearchFilterSort/SearchFilterSort";
+import PlantCard from "@/components/PlantCard/PlantCard";
+import Loading from "@/components/Loading/Loading";
+import Error from "@/components/Error/Error";
+import Link from "next/link";
+import { useRouter } from "next/navigation";
 
-export default function Home() {
+export default function HomePage() {
   const [plants, setPlants] = useState([]);
+  const [filteredPlants, setFilteredPlants] = useState([]);
+  const [loading, setLoading] = useState(true);
+  const [error, setError] = useState(null);
+  const router = useRouter();
 
-  {
-    /*taking the data of plants and storing them in useState*/
-  }
   useEffect(() => {
-    let url = "http://localhost:5000/plants";
-
-    fetch(url)
-      .then((res) => res.json())
-      .then(setPlants);
+    const fetchPlants = async () => {
+      setLoading(true);
+      try {
+        const res = await fetch(`${process.env.NEXT_PUBLIC_API_URL}/plants`);
+        if (!res.ok) throw new Error("Failed to fetch plants");
+        const data = await res.json();
+        const formatted = (data.plants || []).map((p) => ({
+          plantId: p.plantId || p.id,
+          plantName: p.plantName || p.name || "",
+          category: (p.category || p.plantCategory || "").toLowerCase(),
+          price: Number(p.plantPrice || p.price || 0),
+          average_rating: Number(p.average_rating || 0),
+          plantDescription: p.plantDescription || "",
+          plantImage: p.plantImage || "",
+        }));
+        setPlants(formatted);
+      } catch (err) {
+        setError(err);
```

Comment:
The error handling sets the error object directly, which may not always have a message property. Consider normalizing the error before setting state.

Suggested change:

-         setError(err);
+         setError({ message: err?.message || String(err) });
pankaj28843now